doc/source/data/key-concepts.rst

```diff
@@ -0,0 +1,131 @@
+.. _data_key_concepts:
+
+Key Concepts
+============
+
+
+Datasets and blocks
+-------------------
+
+There are two main concepts in Ray Data:
+
+* Datasets
+* Blocks
+
+`Datasets` is the main user-facing Python API. They represent a distributed data collection and define data loading and processing operations. Users typically use the API by:
+
+1. Create a :class:`Dataset <ray.data.Dataset>` from external storage or in-memory data.
+2. Apply transformations to the data.
+3. Write the outputs to external storage or feed the outputs to training workers.
+
+The Dataset API is lazy, meaning that operations aren't executed until you call an action
+like :meth:`~ray.data.Dataset.show`. This allows Ray Data to optimize the execution plan
+and execute operations in parallel.
+
+*Blocks* are the basic unit of data that Ray Data operates on. A block is a contiguous
+subset of rows from a dataset.
```

Comment:
```suggestion
To parallelize processing every `Dataset` is split into `Blocks` -- a subset of rows distributed and processed independently.
```

Reply:
The original version sounds more natural. We can add a sentence like "Blocks are distributed and processed across a Ray cluster independently" if we want